src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -66,66 +56,55 @@ impl ManagedBarrierState {
     /// Create a barrier manager state. This will be called only once.
     pub(super) fn new() -> Self {
         Self {
-            inner: ManagedBarrierStateInner::Pending {
-                // TODO: specify last epoch
-                last_epoch: None,
-            },
+            map: HashMap::new(),
             create_mview_progress: Default::default(),
         }
     }
 
-    fn inner_mut(&mut self) -> &mut ManagedBarrierStateInner {
-        &mut self.inner
-    }
-
     /// Notify if we have collected barriers from all actor ids. The state must be `Issued`.
-    fn may_notify(&mut self) {
-        let (epoch, to_notify) = match self.inner_mut() {
-            ManagedBarrierStateInner::Issued {
-                epoch,
-                remaining_actors,
-                ..
-            } => (*epoch, remaining_actors.is_empty()),
-
+    fn may_notify(&mut self, curr_epoch: u64) {
+        let to_notify = match self.map.get(&curr_epoch) {
+            Some(ManagedBarrierStateInner::Issued {
+                remaining_actors, ..
+            }) => (remaining_actors.is_empty()),
             _ => unreachable!(),
         };
 
         if to_notify {
-            let state = std::mem::replace(
-                self.inner_mut(),
-                ManagedBarrierStateInner::Pending {
-                    last_epoch: Some(epoch),
-                },
-            );
+            let inner = self.map.remove(&curr_epoch);
```

Comment:
We may directly `unwrap` here.
```suggestion
            let inner = self.map.remove(&curr_epoch).unwrap();
```

Reply:
okoko